### ISSUE TITLE
Gelbooru won't show any more images after doing something very specific

### ISSUE THREAD

**Issue by Babyforce** [Bionus/imgbrd-grabber#1467]
**Bug description**

Whenever I browse Gelbooru, there are cases where Grabber says "No result" after doing a specific action. The only way to make it work again is to close and re-open the software. The tag list is still displayed, though.

**Steps to reproduce**

1. Select Gelbooru as source imageboard.
2. Click on OK to start searching; the images display.
3. Enter a tag that doesn't exist and click OK once again, nothing is found.
4. Remove the tag, click OK, and see the result, still nothing is found.

**Expected behavior**

Grabber is supposed to show images from the board after removing the non-existing tag. Entering an existing tag and clicking ok won't solve the issue.

**Context**

I reproduced this bugs a few times with a newly generated settings.ini (Deleting everything in %localappdata%/Bionus/Grabber). This is what is in the main.log when the bug happens with the steps above:

```
[15:12:10.047][Info] Temp directory purged of 0/0 files (0 failed)
[15:12:10.526][Info] New session started.
[15:12:10.527][Info] Software version: 6.0.6.
[15:12:10.527][Info] Path: `C:/Program Files/Grabber_6-0-6`
[15:12:10.527][Info] Loading preferences from `C:/Program Files/Grabber_6-0-6/settings.ini`
[15:12:11.574][Info] SSL libraries: OpenSSL 1.0.2k  26 Jan 2017
[15:12:11.575][Info] Setting language to 'English'...
[15:12:11.577][Info] 17 sources found: anime-pictures.net (Anime pictures), behoimi.org (Danbooru), chan.sankakucomplex.com (Sankaku), danbooru.donmai.us (Danbooru (2.0)), derpibooru.org (Booru-on-rails), e621.net (Danbooru), gelbooru.com (Gelbooru (0.2)), hijiribe.donmai.us (Danbooru (2.0)), idol.sankakucomplex.com (Sankaku), konachan.com (Moebooru), rule34.paheal.net (Shimmie), rule34.xxx (Gelbooru (0.2)), safebooru.org (Gelbooru (0.2)), sonohara.donmai.us (Danbooru (2.0)), tentaclerape.net (Shimmie), www.zerochan.net (Zerochan), yande.re (Moebooru)
[15:12:11.594][Info] Updating checkboxes.
[15:12:11.596][Info] [gelbooru.com] Logging in...
[15:12:12.018][Info] [gelbooru.com] Login finished: failure.
[15:12:12.024][Info] Updating checkboxes.
[15:12:12.030][Info] Monitoring starting
[15:12:12.031][Info] Monitoring tick
[15:12:12.031][Info] Monitoring finished
[15:12:15.710][Warning] [Qt][default] libpng warning: iCCP: known incorrect sRGB profile
[15:12:17.806][Info] Saving sources...
[15:12:17.807][Info] Updating checkboxes.
[15:12:17.808][Info] Loading results...
[15:12:17.809][Info] [gelbooru.com][Html] Loading page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=&amp;pid=0`
[15:12:17.810][Info] [gelbooru.com][Xml] Loading page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=0&amp;tags=`
[15:12:18.026][Info] [gelbooru.com][Xml] Receiving page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=0&amp;tags=`
[15:12:18.038][Info] [gelbooru.com][Xml] Parsed page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=0&amp;tags=`: 20 images (20), 0 tags (339), 4170205 total (4170205), 208511 pages (208511)
[15:12:18.050][Info] [gelbooru.com][Html] Receiving page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=&amp;pid=0`
[15:12:18.065][Info] [gelbooru.com][Html] Parsed page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=&amp;pid=0`: 42 images (42), 43 tags (43), -1 total (-1), -1 pages (-1)
[15:12:25.228][Info] Loading results...
[15:12:25.229][Info] [gelbooru.com][Html] Loading page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=nonexistingtag&amp;pid=0`
[15:12:25.229][Info] [gelbooru.com][Xml] Loading page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=0&amp;tags=nonexistingtag`
[15:12:25.287][Info] [gelbooru.com][Xml] Receiving page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=0&amp;tags=nonexistingtag`
[15:12:25.288][Info] [gelbooru.com][Xml] Parsed page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=0&amp;tags=nonexistingtag`: 0 images (0), 0 tags (0), -1 total (0), 0 pages (0)
[15:12:25.833][Info] [gelbooru.com][Html] Receiving page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=nonexistingtag&amp;pid=0`
[15:12:25.833][Info] [gelbooru.com][Html] Parsed page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=nonexistingtag&amp;pid=0`: 0 images (0), 0 tags (0), -1 total (-1), -1 pages (-1)
[15:12:29.336][Info] Loading results...
[15:12:29.338][Info] [gelbooru.com][Html] Loading page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=&amp;pid=-42`
[15:12:29.339][Info] [gelbooru.com][Xml] Loading page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=-1&amp;tags=`
[15:12:29.394][Info] [gelbooru.com][Xml] Receiving page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=-1&amp;tags=`
[15:12:29.395][Info] [gelbooru.com][Xml] Parsed page `https://gelbooru.com/index.php?page=dapi&amp;s=post&amp;q=index&amp;limit=20&amp;pid=-1&amp;tags=`: 0 images (0), 0 tags (0), -1 total (0), 0 pages (0)
[15:12:30.603][Info] [gelbooru.com][Html] Receiving page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=&amp;pid=-42`
[15:12:30.618][Info] [gelbooru.com][Html] Parsed page `https://gelbooru.com/index.php?page=post&amp;s=list&amp;tags=&amp;pid=-42`: 42 images (42), 43 tags (43), -1 total (-1), -1 pages (-1)
```

**Screenshots**

![grabber_2018-10-13_15-13-54](https://user-images.githubusercontent.com/1695422/46905917-b7776080-cefa-11e8-9829-12976e3a955d.jpg)

**System information**

- OS: Windows 10
- Grabber version: 6.0.6

**Additional context**

This wasn't happening in the version 6.0.5 of Grabber. It also fixes the issue if I replace the default source to Html instead of Xml, but the loading of the images is not the same. This issue might be a duplicate, but I wanted to make sure that this is seen.

![grabber_2018-10-13_15-17-48](https://user-images.githubusercontent.com/1695422/46905942-2d7bc780-cefb-11e8-91d6-b3c6ee48f680.jpg)
#

**Comment by Bionus** [comment]
Actually, this seems to be the same underlying issue as #1450, as we can see in this screenshot you provided:

![](https://user-images.githubusercontent.com/1695422/46905917-b7776080-cefa-11e8-9829-12976e3a955d.jpg)

The page number drops to "0", so when you later change the tags, the page number doesn't change so Grabber will try to load page "0" of your search, page which doesn't exist.

I could indeed reproduce this bug in 6.0.6, and it seems to have been fixed by 03045fa97312af2b9dfd11491732f56496aa0754 in nightly! So it should be solved in 6.1.0 👌 

The fact that changing to the HTML source fixes the issue is strange though, I'll investigate that.

**Comment by Babyforce** [comment]
Oh, I didn't notice about the page changing to "0", I knew something was similar to his issue but couldn't figure out what. Thank you for investigating. I'll go check the nightly build.
